wordpress-mobile/WordPress-Android
# Contact Support: Add Analytics

Fixes #18830 

This PR introduces new events for the chatbot feature. 

#### Events
- **support_chatbot_started(chat_id):** Sent when the chatbot screen launched.
- **support_chatbot_webview_error(chat_id):** Sent when a webview error is received. I used `ErrorManagedWebViewClientListener.onWebViewReceivedError()` function to listen to these errors. It's important to note that this event doesn't cover all types of errors, such as the lack of internet connectivity. I didn't want to overthink what errors we really need for analytics purposes, and I think this is a non-critical event.
- **support_chatbot_ticket_success(chat_id)**: Sent when a ZD ticket is created successfully. Although I attempted to retrieve the ticket number, I couldn't get it from `zendesk.support.Request`. I think this is by design in the ZD API, as we are creating the ticket on behalf of the the user, so the user can't be aware of the ticket id as it is company-specific data.
- **support_chatbot_ticket_failure(error_message):** Sent when a ticket creation fails.
- **support_chatbot_ended(chat_id):** Sent when the chatbot screen is closed.

> **Note**
> I considered moving `ZendeskHelper` to the ViewModel, however, this wouldn't be appropriate because `ZendeskHelper` functions need `context.`

_You need to check Logcat to test events. So, it's preferred to build the app on Android Studio, or you can use alternative ways such as checking them on Tracks (an internal tool)._
**To test success cases:** 
1. Launch the JP app.
2. If the flag of `contact_support` isn't enabled, enable it from "Me → Debug settings".
3. Navigate to "Me → Help → Contact Support".
4. Check the logcat and verify `🔵 Tracked: support_chatbot_started, Properties: {"chat_id":"2bb027ff-8f26-455b-989c-935969e55861"}` was displayed. _(chat_id will be random)_
5. Ask a question to the bot.
6. Tap "Contact support" button.
7. Verify `🔵 Tracked: support_chatbot_ticket_success, Properties: {"chat_id":"2bb027ff-8f26-455b-989c-935969e55861"}` was displayed.
8. Verify `🔵 Tracked: support_chatbot_ended, Properties: {"chat_id":"2bb027ff-8f26-455b-989c-935969e55861"}` was displayed.

**To test error cases:** 
1. Launch the JP app.
2. If the flag of `contact_support` isn't enabled, enable it from "Me → Debug settings".
3. Navigate to "Me → Help → Contact Support".
5. Ask a question to the bot.
6. Disable the internet connection by enabling airplane mode.
7. Tap "Contact support".
8. Verify `🔵 Tracked: support_chatbot_ticket_failure, Properties: {"chat_id":"b6c37e9a-7bd9-47f1-893b-c3558b6b9d89","error_message":"Unable to resolve host \"***.zendesk.com\": No address associated with hostname"}` was displayed.

I couldn't reproduce `support_chatbot_webview_error` event. I think that error should be triggered from JS side, and it should be an error from the bot.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Nothing.

3. What automated tests I added (or what prevented me from doing so)
This PR doesn't include any UI changes. So, adding UI tests doesn't make sense. Unit test for `SupportWebViewActivityViewModel` could be added, however, the logic in the class is too simple, just sending analytics events when class functions are called. So, I forgo adding any tests.

PR submission checklist:


UI Changes testing checklist:
